metadata/com.github.luben/zstd-jni/1.5.2-5/jni-config.json

```diff
@@ -0,0 +1,74 @@
+[
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdCompressCtx"
+    },
+    "name": "com.github.luben.zstd.ZstdCompressCtx",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDecompressCtx"
+    },
+    "name": "com.github.luben.zstd.ZstdDecompressCtx",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDictCompress"
+    },
+    "name": "com.github.luben.zstd.ZstdDictCompress",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdDictDecompress"
+    },
+    "name": "com.github.luben.zstd.ZstdDictDecompress",
+    "fields": [
+      {
+        "name": "nativePtr"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdInputStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdInputStreamNoFinalizer",
+    "fields": [
+      {
+        "name": "dstPos"
+      },
+      {
+        "name": "srcPos"
+      }
+    ]
+  },
+  {
+    "condition": {
+      "typeReachable": "com.github.luben.zstd.ZstdOutputStreamNoFinalizer"
+    },
+    "name": "com.github.luben.zstd.ZstdOutputStreamNoFinalizer",
+    "fields": [
+      {
+        "name": "dstPos"
+      },
+      {
+        "name": "srcPos"
+      }
+    ]
+  }
+]
```

Comment:
May be we can add in similar manner support for `ZstdDirectBuffer(Compress|Decompress)NoFinalizer`, the fields there are named `produced` and `consumed`.

Reply:
- I'll see how to add corresponding unit tests for this.

Reply:
- I can only seem to see use cases for this class at https://github.com/luben/zstd-jni/blob/v1.5.2-3/src/test/scala/Zstd.scala#L303 , so I guess by In the way of scala grammatical meaning, a unit test corresponding to java has been written.